### PR TITLE
Common transform manipulation

### PR DESCRIPTION
**Pull request by Wardergrip**

### CODE REVIEW

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/TransformMimic.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        [SerializeField] private bool _mimicPosition = true;
+        [SerializeField] private bool _mimicRotation = true;
+
+        void Update()
+        {
+            if (_transformToMimic == null) return;
+
+            if (_mimicPosition)
+            {
+                transform.position = _transformToMimic.position;
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/TransformMimic.cs]: Maybe add an offset as well so you can easily have e.g. a camera follow a player or object using a set offset

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        [SerializeField] private bool _flipAroundY = false;
+
+        private Vector3 _lookPos = new();
+        private bool _isInitialized = false;
+
+        void Start()
+        {
+            if (_cameraObject == null)
+            {
+                _camera = FindAnyObjectByType<Camera>();
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs]: Maybe we should check for Camera.main as well?

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/TransformMimic.cs (hunk 1 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/TransformMimic.cs]: That is indeed a good suggestion

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs (hunk 2 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs]: Good point!

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                if (_camera == null)
+                {
+                    _camera = Camera.main;
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs]: This check needs to be first. There will never be a Camera.main if FindAnyObjectByType returns null.

**Review by AtlantiaKing — CHANGES_REQUESTED** [review]

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs (hunk 3 above)
Comment (Wardergrip) [reply ThatDefaultProject/Assets/ThatDefaultProject/Scripts/LookAtCamera.cs]: Right, it does make more sense to prioritize Camera.main over any camera

**Review by AtlantiaKing — APPROVED** [review]

**Pull request merged**